kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/config/EncryptionBufferConfig.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.config;
+
+/**
+ *
+ * @param minSizeBytes the minimum size of the encryption buffer
+ * @param maxSizeBytes the maximum size of the encryption buffer
+ */
+public record EncryptionBufferConfig(int minSizeBytes, int maxSizeBytes) {
```

Comment:
I've named it min size rather than initial as I'm thinking of making the initial sizing logic smarter in a subsequent PR.

We could inspect the incoming memory records and use it's size in bytes plus an approximate overhead per message to make a better guess at the initial size requirement (noting that the overhead is KMS dependent so it would be a fudge factor rather than something accurate). We would use `min(maxBufSize, max(minBufSize, approximateOutputSize)` as the initial buffer size.